source/actors/ActionBus.gd

```diff
@@ -0,0 +1,61 @@
+extends Node
+
+"""
+Manages and signalize Action input events
```

Comment:
I am afraid the usage of signals, especially through the interface, can hide some side effects, or even make debugging harder...maybe even spaghettify the logic...

Reply:
It would be cool knowing which signal is attached to which module. But I actually love the idea of signals, it will be useful if you change the implementation on the module who send the signal or receive it. 
Still I am wrapping my head around it

Reply:
on the topic, I find easier in order to debug two things: 
1. Connect signals from code and not from editor
2. Call the functions triggered by signal `on_<NodeWhoEmitSignal>_<signal_name>` in this way you can find easily who would emit which signall

Reply:
Yeah I thought about following this convention, `_on_<EmitterNam>_<signal_name>`, but it will kinda pollute the code, the movement modules are not meant to work only with these signals. Unless I add an intermediary layer to connect and handle these signals from code. Having something like a middle man from which the signals will be connected to, and will also trigger the modules methods.

From inspector is not hard to track the signals connection tho, but my concern is further debug.

![image](https://user-images.githubusercontent.com/10171059/57394791-c70f0d00-719c-11e9-8f00-bc02b08f5788.png)

Wachu think about the intermediary class to mediate the signals?

Reply:
Well I will merge as is, throught the inspector is kinda visible where they are coming, and with the [signals connections being displayed in the GDScript](https://github.com/godotengine/godot/pull/28234) soon this can be easier to track...